### PR TITLE
Adding movies list component and also small changes to the old components(Button,PopOver,MovieCard)

### PR DESCRIPTION
**Pull request by auqibyousuf**

### CODE REVIEW

File: tailwind.config.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ const config: Config = {
         'gradient-conic':
           'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
         linearGradient: 'linear-gradient(90deg, #1ED5A9 0%, #01B4E4 100%)',
+        buttonTab: 'linear-gradient(90deg, #C0FECF 0%, #1ED5A9 100%)',
```
Comment (UmmarHamid) [inline tailwind.config.ts]: we can do gradients using classes easily. do you see this buttonTab gradient reusable?
If not then we should fallback to using gradients with classes like bg-gradient-to-r  from-lightGreen to-lightBlue

**Review by UmmarHamid — COMMENTED** [review]
Can you add a SS so that I can review better?

File: tailwind.config.ts (hunk 1 above)
Comment (auqibyousuf) [reply tailwind.config.ts]: did not implemented gradient i thought it will work on active button so I left it will remove this config

**Comment by auqibyousuf** [comment]
Screenshot 2023-10-06 at 6 31 07 PM

File: app/components/Button/Button.tsx [hunk 2]
```diff
@@ -2,15 +2,24 @@ import React from 'react'
...
+  variant,
+  extraClasses,
+  value,
+  onClick,
+}: ButtonType) => {
   const buttonClasses = classNames({
     'py-3 px-6 rounded-30 bg-linearGradient text-white px-7 py-3  w-28 h-12 flex items-center justify-center':
       variant == 'hero-section',
+    'px-5 py-1 rounded-5 text-darkBlue text-[16px] text-semibold':
+      variant == 'tabs',
```
Comment (UmmarHamid) [inline app/components/Button/Button.tsx]:
```suggestion
      variant == 'tab',
```

File: app/components/MovieCard/MovieCardTypes.ts [hunk 3]
```diff
@@ -12,5 +12,5 @@ export type MovieCardTypes = {
   onCardClick?: React.MouseEventHandler
   onMenuClick?: React.MouseEventHandler
   extraClasses?: string
-  menuPoplinks: MenuPopOverTypes
+  menuPopLinks: MenuPopOverTypes
```
Comment (UmmarHamid) [inline app/components/MovieCard/MovieCardTypes.ts]:
```suggestion
  menuPopOverLinks: MenuPopOverTypes
```

File: app/components/MovieList/MovieListTypes.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { ButtonType } from '../Button/ButtonTypes'
+import { MovieCardTypes } from '../MovieCard/MovieCardTypes'
+
+export type MovieListTypes = {
+  movies: Array<MovieCardTypes>
+  buttonTabs: Array<ButtonType>
```
Comment (UmmarHamid) [inline app/components/MovieList/MovieListTypes.ts]:
```suggestion
tabs?: Array<ButtonType>
```

File: app/components/MovieList/MovieList.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                rating={movie.rating}
+                title={movie.title}
+                menuPopLinks={movie.menuPopLinks}
+                date={movie.date}
+              />
+            </li>
+          )
+        })}
+      </ul>
+    </div>
```
Comment (UmmarHamid) [inline app/components/MovieList/MovieList.tsx]:
```suggestion
    <div className='pl-10'>
      <div className='flex gap-5 mt-8'>
        <Heading text={title} variant='24' />
        {tabs.length>0 &&
        <div className={buttonTabClasses}>
          {buttonTabs.map((buttonTab, index) => {
            return (
              <Button
                key={index}
                text={buttonTab.text}
                variant='tabs'
                onClick={buttonTab.onClick}
                value={buttonTab.text}
              />
            )
          })}
        </div>
        }
      </div>

      <ul className={`${moviesListClasses} ${extraClasses ?? ''}`}>
        {movies.map((movie, index) => {
          return (
            <li key={index} className='flex shrink-0 basis-40'>
              <MovieCard
                cardImage={movie.cardImage}
                rating={movie.rating}
                title={movie.title}
                menuPopLinks={movie.menuPopLinks}
                date={movie.date}
              />
            </li>
          )
        })}
      </ul>
    </div>
```

**Review by UmmarHamid — CHANGES_REQUESTED** [review]
Nice work!
Small feedback

**Review by UmmarHamid — APPROVED** [review]

**Pull request merged**